### PR TITLE
tests: setup codecov

### PR DESCRIPTION
**Pull request by dapplion**
Completes remaining task from https://github.com/aragon/radspec/pull/96

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,16 +8,20 @@ jobs:
...
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Run linter
+        run: yarn lint
+      - name: Run tests
+        run: yarn test:coverage
+        env:
```
Comment (sohkai) [inline .github/workflows/ci.yml:22]: I wonder if we should move this to be outside of this particular step so the entire job runs with `CI: true`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,16 +8,20 @@ jobs:
...
-        node-version: ${{ matrix.node-version }}
-    - name: Install dependencies
-      run: yarn install --frozen-lockfile
-    - name: Run linter
-      run: yarn lint
-    - name: Run tests
-      run: yarn test
-    env:
-      CI: true
+      - uses: actions/checkout@v1
```
Comment (sohkai) [inline .github/workflows/ci.yml:11]: Could we leave the indentation on the same level? There's no difference in yaml (the `-` is already treated like an indentation), and better to leave our action yaml looking like what's in the Github documentation :).

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dapplion) [reply .github/workflows/ci.yml:22]: Sure! I wasn't sure where that was supposed to be

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dapplion) [reply .github/workflows/ci.yml:11]: My .yaml linter recommended using this indentation. If you look closely the Github Actions documentation is very inconsistent regarding array indentation, take a look at different sections in this page https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions

Looking at various sources seems that this indentation is more popular https://yaml.org/start.html https://docs.docker.com/compose/compose-file/

**Comment by dapplion** [comment]
> Couple of notes, but thanks for doing this!
> 
> Ah, I guess we ran into the same thing on the buidler plugin—I didn't realize that the `push` event wasn't running on forked pull requests (I guess those are covered by the `pull-request` event). Could we also add that to the `on` list?

Sure! Added in the last commit

File: .github/workflows/ci.yml (hunk 2 above)
Comment (sohkai) [reply .github/workflows/ci.yml:14]: Hmm good point. I spent a bit of time aligning them all to be on the same level today... but it looks like I should revert and make it consistent against the indented version.

**Pull request merged**